### ISSUE TITLE
.pylintrc not used

### ISSUE THREAD

**Issue by ai-learner-00** [microsoft/vscode-pylint#569]
## Diagnostic Data
- Python version (& distribution if applicable, e.g., Anaconda): Python 3.10
- Type of virtual environment used (e.g., conda, venv, virtualenv, etc.): Dev container
- Operating system (and version): Windows
- Version of tool extension you are using: 2023.11.13481007

### Expected Behavior

I added a `.pylintrc` to ignore the tests directory.
```
[MASTER]
ignore=tests
```

### Actual Behavior

I still get warnings for a test file.
![image](https://github.com/user-attachments/assets/71b4ee4e-3ecd-4705-bb98-3c8e31b485ec)

I tried specifying the rcfile, but it didn't work.
```
				"pylint.enabled": true,
				"pylint.args": [
					"--rcfile=.pylintrc"
				],
```
Also, I tried `"pylint.enabled": false` to see if settings were used, and the warnings still appeared.

## Reproduction Steps:

[pyproject.zip](https://github.com/user-attachments/files/16678449/pyproject.zip)
1. Open the following code, which installs the pylint extension, in a dev container
2. Go to `test_postal_codes.py` to see if there are warnings.

## Logs:

<details>
  <summary>Click here for detailed logs</summary>

2024-08-20 16:00:31.613 [info] [Trace - 4:00:31 PM] Sending request 'textDocument/codeAction - (27)'.
2024-08-20 16:00:31.613 [info] Params: {
    "textDocument": {
        "uri": "file:///workspaces/Service-GreatExpectations/tests/unit/custom_expectations/test_postal_codes.py"
    },
    "range": {
        "start": {
            "line": 0,
            "character": 0
        },
        "end": {
            "line": 0,
            "character": 0
        }
    },
    "context": {
        "diagnostics": [
            {
                "range": {
                    "start": {
                        "line": 0,
                        "character": 0
                    },
                    "end": {
                        "line": 0,
                        "character": 0
                    }
                },
                "message": "Missing module docstring",
                "code": "C0114:missing-module-docstring",
                "codeDescription": {
                    "href": "https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/missing-module-docstring.html"
                },
                "severity": 3,
                "source": "Pylint"
            },
            {
                "range": {
                    "start": {
                        "line": 0,
                        "character": 0
                    },
                    "end": {
                        "line": 0,
                        "character": 36
                    }
                },
                "message": "Missing function or method docstring",
                "code": "C0116:missing-function-docstring",
                "codeDescription": {
                    "href": "https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/missing-function-docstring.html"
                },
                "severity": 3,
                "source": "Pylint"
            }
        ],
        "triggerKind": 2
    }
}

2024-08-20 16:00:31.625 [info] [Trace - 4:00:31 PM] Received response 'textDocument/codeAction - (27)' in 12ms.
2024-08-20 16:00:31.625 [info] Result: []

2024-08-20 16:00:31.809 [info] [Trace - 4:00:31 PM] Sending request 'textDocument/codeAction - (28)'.
2024-08-20 16:00:31.809 [info] Params: {
    "textDocument": {
        "uri": "file:///workspaces/Service-GreatExpectations/tests/unit/custom_expectations/test_postal_codes.py"
    },
    "range": {
        "start": {
            "line": 2,
            "character": 0
        },
        "end": {
            "line": 2,
            "character": 0
        }
    },
    "context": {
        "diagnostics": [],
        "triggerKind": 2
    }
}

2024-08-20 16:00:31.811 [info] [Trace - 4:00:31 PM] Received response 'textDocument/codeAction - (28)' in 2ms.
2024-08-20 16:00:31.811 [info] Result: []

2024-08-20 16:00:32.500 [info] [Trace - 4:00:32 PM] Sending request 'textDocument/codeAction - (29)'.
2024-08-20 16:00:32.501 [info] Params: {
    "textDocument": {
        "uri": "file:///workspaces/Service-GreatExpectations/tests/unit/custom_expectations/test_postal_codes.py"
    },
    "range": {
        "start": {
            "line": 1,
            "character": 35
        },
        "end": {
            "line": 1,
            "character": 35
        }
    },
    "context": {
        "diagnostics": [],
        "triggerKind": 2
    }
}

2024-08-20 16:00:32.502 [info] [Trace - 4:00:32 PM] Received response 'textDocument/codeAction - (29)' in 2ms.
2024-08-20 16:00:32.502 [info] Result: []

</details>

## Outcome When Attempting Debugging Steps:

Did running it from the command line work? With `pylint tests`, there are no warnings.

**Comment by karthiknadig** [comment]
@ai-learner-00 This is because we explicitly ask for linting on a file. When you run pylint on a specific file, it does not use ignore. You can use `pylint.ignorePatterns` to tell the extension to not run linter on such files.

you can try this manually, run `pylint tests\test_postal_codes.py` and it should lint it.
![image](https://github.com/user-attachments/assets/4fc7b63f-b9cf-45b3-9b2a-3bd36b6b9e2a)
